.github/workflows/full_build.yml

```diff
@@ -36,118 +35,244 @@ jobs:
       with:
         submodules: 'recursive'
 
-    # To get checks on Pull requests, see https://github.com/LouisBrunner/checks-action
-    # - uses: LouisBrunner/checks-action@v1.1.1
-    #   if: always()
-    #   with:
-    #     token: ${{ secrets.GITHUB_TOKEN }}
-    #     name: Test Full Build
-    #     conclusion: ${{ job.status }}
-    #     # output: |
-    #     #   {"summary":${{ steps.test.outputs.summary }}}
-
     - name: Create Build Environment
-      # Some projects don't allow in-source building, so create a separate build directory
-      # We'll use this as our working directory for all subsequent commands
       working-directory: ${{runner.workspace}}
-      run: cmake -E make_directory ${{runner.workspace}}/build
+      run: cmake -E make_directory ${{runner.workspace}}/acados/build
 
-
-    - name: Install Eigen (C test dependency)
+    - name: Install Eigen (ctest dependency)
       working-directory: ${{runner.workspace}}/acados
       shell: bash
       # TODO: cache
       run: ./ci/shared/install_eigen.sh
 
-
     - name: Configure CMake
-      # Use a bash shell so we can use the same syntax for environment variable
-      # access regardless of the host operating system
       shell: bash
-      working-directory: ${{runner.workspace}}/build
+      working-directory: ${{runner.workspace}}/acados/build
       # Note the current convention is to use the -S and -B options here to specify source
       # and build directories, but this is only available with CMake 3.13 and higher.
       # The CMake binaries on the Github Actions machines are (as of this writing) 3.12
-      run: cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DACADOS_WITH_QPOASES=$ACADOS_WITH_QPOASES -DACADOS_WITH_DAQP=$ACADOS_WITH_DAQP  -DACADOS_WITH_QPDUNES=$ACADOS_WITH_QPDUNES -DACADOS_WITH_OSQP=$ACADOS_WITH_OSQP -DACADOS_PYTHON=$ACADOS_PYTHON -DACADOS_UNIT_TESTS=$ACADOS_UNIT_TESTS -DACADOS_OCTAVE=$ACADOS_OCTAVE
-
+      run: cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DACADOS_WITH_QPOASES=$ACADOS_WITH_QPOASES -DACADOS_WITH_DAQP=$ACADOS_WITH_DAQP  -DACADOS_WITH_QPDUNES=$ACADOS_WITH_QPDUNES -DACADOS_WITH_OSQP=$ACADOS_WITH_OSQP -DACADOS_PYTHON=$ACADOS_PYTHON -DACADOS_OCTAVE=OFF
 
     - name: Build & Install
-      working-directory: ${{runner.workspace}}/build
+      working-directory: ${{runner.workspace}}/acados/build
       shell: bash
       run: |
         cmake --build . --config $BUILD_TYPE
         make install -j4
 
-    - name: Export Paths Lib
+    - name: Store shared libs (/lib)
+      uses: actions/upload-artifact@v4
+      with:
+        name: lib
+        if-no-files-found: error
+        path: ${{runner.workspace}}/acados/lib/
+        compression-level: 0 #TODO test time difference
+        overwrite: true
+
+    - name: Store build scripts (/build)
```

Comment:
Is storing the `build` folder needed?

Reply:
yes, the ctest cmake file is defined there